File: src/896. Monotonic Array/main.ts
// Solved by @kitanoyoru
// https://leetcode.com/problems/monotonic-array/


const isMonotonic = (nums: number[]): boolean => {
  let incCount = 0, decrCount = 0;

  for (let i = 1; i < nums.length; i++) {
    if (nums[i] > nums[i-1]) {
      incCount++;
    } else if (nums[i] < nums[i-1]) {
      decrCount++;
    }
  }

  return !(incCount && decrCount);
}

console.log(isMonotonic([1, 1, 0]));

File: src/896. Monotonic Array/main.py
# Solved by @kitanoyoru
# https://leetcode.com/problems/monotonic-array/


class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        inc, dec = True, True
        for i in range(1, len(nums)):
            inc &= nums[i] <= nums[i - 1]
            dec &= nums[i] >= nums[i - 1]
        return inc or dec
